sustainml_cpp/include/sustainml_cpp/nodes/HardwareConstraintsNode.hpp

```diff
@@ -0,0 +1,128 @@
+// Copyright 2024 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file HardwareConstraintsNode.hpp
+ */
+
+
+#ifndef SUSTAINMLCPP_NODES_HARDWARECONSTRAINTSNODE_HPP
+#define SUSTAINMLCPP_NODES_HARDWARECONSTRAINTSNODE_HPP
+
+#include <sustainml_cpp/core/Node.hpp>
+#include <sustainml_cpp/core/Callable.hpp>
+#include <sustainml_cpp/types/types.h>
+
+#include <vector>
+#include <memory>
+
+namespace sustainml {
+namespace core {
+template<class T> class QueuedNodeListener;
+} // namespace core
+
+namespace utils {
+template<class T> class SamplePool;
+} // namespace utils
+
+namespace hw_constraints_module {
```

Comment:
We may think of using the same namespace as for the hwResourcesNode, the `hw_module`